ee/se/modules/380-metallb/hooks/discovery_l2_load_balancers.go

```diff
@@ -123,50 +113,36 @@ func applyLoadBalancerFilter(obj *unstructured.Unstructured) (go_hook.FilterResu
 	}
 
 	interfaces := []string{}
-	if len(l2loadbalancer.Spec.Interfaces) > 0 {
-		interfaces = l2loadbalancer.Spec.Interfaces
+	if len(l2loadbalancer.Spec.L2.Interfaces) > 0 {
+		interfaces = l2loadbalancer.Spec.L2.Interfaces
 	}
 
 	return L2LoadBalancerInfo{
 		Name:         l2loadbalancer.Name,
 		AddressPool:  l2loadbalancer.Spec.AddressPool,
 		Interfaces:   interfaces,
 		NodeSelector: l2loadbalancer.Spec.NodeSelector,
+		IsDefault:    l2loadbalancer.Spec.IsDefault,
 	}, nil
 }
 
 func handleLoadBalancers(input *go_hook.HookInput) error {
-	input.MetricsCollector.Expire("d8_l2_load_balancer")
 	l2lbservices := make([]L2LBServiceConfig, 0, 4)
-	l2loadbalancers := makeL2LoadBalancersMapFromSnapshot(input.Snapshots["l2loadbalancers"])
-	l2lbLoadBalancerClass := input.Values.Get("l2LoadBalancer.loadBalancerClass").String()
+	l2loadbalancers, l2lbDefault := makeL2LoadBalancersMapFromSnapshot(input.Snapshots["l2loadbalancers"])
 
 	for _, serviceSnap := range input.Snapshots["services"] {
 		service, ok := serviceSnap.(ServiceInfo)
 		if !ok {
 			continue
 		}
 
-		if service.L2LBNameAnnotationIsMissed {
-			input.LogEntry.Warnf("Annotation %s is missed for service %s in namespace %s", keyAnnotationL2BalancerName, service.Name, service.Namespace)
-			continue
-		}
-
-		if (l2lbLoadBalancerClass != "" && service.LoadBalancerClass != l2lbLoadBalancerClass) ||
-			(l2lbLoadBalancerClass == "" && service.LoadBalancerClass != "") {
-			// two possible cases:
-			// loadBalancerClass is set in mc and hook watches only services with the same class
-			// loadBalancerClass is empty and hook watches only services without the class
-			continue
-		}
-
-		l2lb, exists := l2loadbalancers[service.L2LoadBalancerName]
+		l2lb, exists := l2loadbalancers[service.LoadBalancerClass]
 		if !exists {
-			// L2LoadBalancer is not found by name
-			input.MetricsCollector.Set("d8_l2_load_balancer_orphan_service", 1, map[string]string{"namespace": service.Namespace, "name": service.Name}, metrics.WithGroup("d8_l2_load_balancer"))
-			continue
+			l2lb = l2loadbalancers[l2lbDefault]
```

Comment:
And if there isn't a default MetalLoadBalancerClass?

Reply:
Done